lib/spack/llnl/util/lang.py

```diff
@@ -291,8 +452,9 @@ def __len__(self):
     def __delitem__(self, key):
         del self.dict[key]
 
-    def _cmp_key(self):
-        return tuple(sorted(self.values()))
+    def _cmp_iter(self):
+        for _, v in sorted(self.items()):
```

Comment:
I assume the ordering is arbitrary for us, but want to note that this is different from the previous one (the ordering is done here on `(key, value)` pairs, while before it was just the values)